containers/src/Kokkos_Bitset.hpp

```diff
@@ -70,13 +87,35 @@ class Bitset {
     block_shift = Kokkos::Impl::integral_power_of_two(block_size)
   };
 
+  //! Type of @ref m_blocks.
+  using block_view_type = View<unsigned*, Device, MemoryTraits<RandomAccess>>;
+
  public:
+  KOKKOS_DEFAULTED_FUNCTION
+  Bitset() = default;
```

Comment:
1. Indeed, I can't remember why I made it default-constructible on the device. I guess I looked at how the default constructor was declared for `Kokkos::View` (which is also declared `KOKKOS_DEFAULTED_FUNCTION`). I don't have any strong opinion on that, so if you deem it useless that it be default-constructible on device, I can remove `KOKKOS_DEFAULTED_FUNCTION`.
2. I think I wanted to have - again - something similar to the `Kokkos::View`. If you don't give it a `arg_size`, default construct everything. Otherwise, initialize (see https://github.com/kokkos/kokkos/pull/6198#discussion_r1310465971 maybe). My point was really to make sure no allocation happens before the user really wants it to happen.

Reply:
2. I guess it also does not make much sense to initialize the `Kokkos::Bitset` why a zero size?

Reply:
Ultimately, this is a drive-by change that changes behavior. Thus, I would prefer to address it in a different pull request.

Reply:
Are you asking just to revert `KOKKOS_DEFAULTED_FUNCTION` for the constructor or more than that? :smile:

Reply:
I would just leave this constructor's signature alone in this pull request.

Reply:
Ok, just to make my mind clear :smile:

So, instead of the current change:
```c++
  KOKKOS_DEFAULTED_FUNCTION
  Bitset() = default;

  /// constructor
  /// arg_size := number of bit in set
  explicit Bitset(unsigned arg_size) : Bitset(Kokkos::view_alloc(), arg_size) {}

  template <class... P>
  Bitset(const Impl::ViewCtorProp<P...>& arg_prop, unsigned arg_size)
```
I should do
```c++
  /// constructor
  /// arg_size := number of bit in set
  explicit Bitset(unsigned arg_size = 0u) : Bitset(Kokkos::view_alloc(), arg_size) {}

  template <class... P>
  Bitset(const Impl::ViewCtorProp<P...>& arg_prop, unsigned arg_size)
```
to stick with the previous behavior (`Bitset` always allocated, even with a zero size) and create an issue to discuss whether or not `Kokkos::Bitset` should be default constructed?

Sorry about the confusion :confused:

Reply:
Hell @masterleinad  ! Please don't forget me :angel:

Reply:
> I should do [...] to stick with the previous behavior (Bitset always allocated, even with a zero size) and create an issue to discuss whether or not Kokkos::Bitset should be default constructed?

Yes.

Reply:
* For the issue, there you go: #6467